Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -4811,21 +4883,31 @@ public static string GetDescription(string ControlAction, string LocateValue, st
             return "Set Web Element '" + LocateValue + "'";
         }
 
-        public static ActUIElement.eElementAction GetUIElementAction(string ControlAction)
+        public static eElementType GetUIElementTypeByPayLoadType(string Type)
         {
-            switch (ControlAction)
+            switch (Type.ToLower())
             {
-                case "Click":
-                    return ActUIElement.eElementAction.Click;
-
-                case "SetValue":
-                    return ActUIElement.eElementAction.SetValue;
-
-                case "SendKeys":
-                    return ActUIElement.eElementAction.SendKeys;
+                case "submit":
+                case "button":
+                    return eElementType.Button;
+                case "text":
+                    return eElementType.TextBox;
+                    // return eElementType.Text; // ???
+                case "textarea":
+                    return eElementType.TextBox;
+                case "select-one":
+                    return eElementType.ComboBox;
+                case "checkbox":
+                    return eElementType.CheckBox;
+                case "radio":
+                    return eElementType.RadioButton;
+                case "span":
+                    return eElementType.Span;
+                case "li":
+                    return eElementType.List;
+                default:
+                    return eElementType.Unknown;
             }
-
-            return ActUIElement.eElementAction.WaitUntilDisplay;  // ??????
         }
 
```

Comment:
GetUIElementTypeByPayLoadType() method need to be removed when change the solution so JS will return the exact name of Element type